native/src/components/MapView.tsx

```diff
@@ -12,49 +14,84 @@ const MapContainer = styled.View`
 const StyledMap = styled(MapboxGL.MapView)`
   width: 700px;
   height: 500px;
+  z-index: 1;
 `
 
 type MapViewPropsType = {
   boundingBox: BBox
   featureCollection: FeatureCollection
+  currentFeature?: Feature<Point>
 }
 
 const textOffsetY = 1.25
-const layerProps: SymbolLayerProps = {
-  id: 'point',
-  style: {
-    symbolPlacement: 'point',
-    iconAllowOverlap: true,
-    iconIgnorePlacement: true,
-    iconImage: ['get', 'symbol'],
-    textField: ['get', 'title'],
-    textFont: ['Roboto Regular'],
-    textOffset: [0, textOffsetY],
-    textAnchor: 'top',
-    textSize: 12
-  }
-}
+const layerId = 'point'
 
 // Has to be set even if we use map libre
 MapboxGL.setAccessToken(mapConfig.accessToken)
-const MapView = ({ boundingBox, featureCollection }: MapViewPropsType): ReactElement => {
+const MapView = ({ boundingBox, featureCollection, currentFeature }: MapViewPropsType): ReactElement => {
+  const mapRef = React.useRef<MapboxGL.MapView | null>(null)
+  const cameraRef = React.useRef<MapboxGL.Camera | null>(null)
+  const [activeFeature, setActiveFeature] = useState<Feature<Point> | null>(currentFeature ?? null)
+  const [showPopup, setShowPopup] = useState<boolean>(!!currentFeature)
```

Comment:
Do we need this in the state? Can't we just set the active feature to null if the popup is closed?

Reply:
no because the popup has to set true auto when redirecting from details page

Reply:
You sure? It is only ever updated together with activeFeature and its initial value also depends on it.

Reply:
Just tried it, seems to work without it as well.

Reply:
word.. :)